src/Generator.cpp

```diff
@@ -1047,27 +1050,49 @@ void execute_generator(const ExecuteGeneratorArgs &args_in) {
     if (!args.generator_name.empty()) {
         const std::string base_path = args.output_dir + "/" + args.file_base_name;
         debug(1) << "Generator " << args.generator_name << " has base_path " << base_path << "\n";
+
+        // Factory function for creating a generator instance for a given function name and target
+        auto generator_factory = [&](const std::string &function_name, const Target &target) -> AbstractGeneratorPtr {
+            // Must re-create each time since each instance will have a different Target.
+            auto gen = args.create_generator(args.generator_name, GeneratorContext(target));
+            for (const auto &kv : args.generator_params) {
+                if (kv.first == "target") {
+                    continue;
+                }
+                gen->set_generatorparam_value(kv.first, kv.second);
+            }
+            return gen;
+        };
+
         if (args.output_types.count(OutputFileType::cpp_stub)) {
-            // When generating cpp_stub, we ignore all generator args passed in, and supply a fake Target.
+            // When generating cpp_stub we ignore all generator args passed in, and supply a fake Target.
             // (CompilerLogger is never enabled for cpp_stub, for now anyway.)
             const Target fake_target = Target();
             auto gen = args.create_generator(args.generator_name, GeneratorContext(fake_target));
             auto output_files = compute_output_files(fake_target, base_path, args.output_types);
-            gen->emit_cpp_stub(output_files[OutputFileType::cpp_stub]);
+            if (args.output_types.count(OutputFileType::cpp_stub)) {
+                gen->emit_cpp_stub(output_files[OutputFileType::cpp_stub]);
+            }
+        }
+
+        if (args.output_types.count(OutputFileType::hlpipe)) {
+            // When serializing a halide pipeline, target is required (since the schedule may be target dependent)
```

Comment:
These are complete sentences, so they really should end with proper punctuation.

Reply:
Okay, will fix!